primeqa/qg/run_qg.py

```diff
@@ -184,23 +184,42 @@ def main(raw_args):
 
     qg_model = QGModel(model_args.model_name_or_path, modality=model_args.modality)
 
+
+
     qgdl = QGDataLoader(
         tokenizer=qg_model.tokenizer,
         dataset_name=data_args.dataset_name,
         modality=model_args.modality,
         input_max_len=data_args.max_len,
         target_max_len=data_args.target_max_len,
     )
-    if training_args.do_train or training_args.do_eval:
-        train_dataset = qgdl.create(
-            dataset_split="train", dataset_config=data_args.dataset_config
-        )
-        valid_dataset = qgdl.create(
-            dataset_split="validation", dataset_config=data_args.dataset_config
-        )
 
-        compute_metrics = rouge_metrics(qg_model.tokenizer)
+    train_dataset = None
+    valid_dataset = None
+    if data_args.train_file is not None:
+        dataset = load_dataset("json", data_files=data_args.train_file)
+        dataset = dataset['train']
+        train_dataset = qgdl.create(dataset=dataset)
+    if data_args.eval_file is not None:
+        dataset = load_dataset("json", data_files=data_args.eval_file)
+        # this is not a bug, by default huggingface datasets library loads any data as train split
+        dataset = dataset['train']
+        valid_dataset = qgdl.create(dataset=dataset)
 
+    if training_args.do_train:
+        if data_args.train_file is None:
+            train_dataset = qgdl.create(
+                dataset_split="train", dataset_config=data_args.dataset_config
+            )
```

Comment:
Actually, having the `data_args.train_file is not None` condition (from line 199) as an `else` here is probably the cleanest option.

Reply:
@ma-sultan : thanks for the suggestion, I made the changes.